src/Native/Unix/System.Native/pal_io.c

```diff
@@ -49,6 +49,25 @@ extern int     getpeereid(int, uid_t *__restrict__, gid_t *__restrict__);
 extern ssize_t  getline(char **, size_t *, FILE *);
 #endif
 
+#if defined(__linux__)
+#if defined(__x86_64__)
+#define PAL_COPY_FILE_RANGE_SYSCALL 326
+#elif defined(__i386__)
+#define PAL_COPY_FILE_RANGE_SYSCALL 377
+#elif defined(__arm__)
+#define PAL_COPY_FILE_RANGE_SYSCALL 391
+#elif defined(__aarch64__)
+#define PAL_COPY_FILE_RANGE_SYSCALL 285
+#else
+#    include <syscall.h>
```

Comment:
I have some questions, help me understand these definitions.

1. Why not do the calls inside the final `else` at the beginning? If `syscall.h` has `__NR_copy_file_range` defined, then you could use that value. If it's not found then you could do something else: either defining them yourself one by one depending on the arch (as you're currently doing), or maybe import them from somewhere else (see question 2).
2. While searching for other usage examples of `copy_file_range`, I found that a file called `systemd/missing_syscall.h` exists, in which these same values are defined for each architecture. Judging by the name of the file, its purpose is probably to be imported when `syscall.h` is not found. If you import this, then you can use the `PAL_COPY_FILE_RANGE_SYSCALL` values defined there, instead of redefining.
3. Also, where are these numbers defined for each architecture?

Reply:
I can do (1), but it's kind of a po-tay-to/po-tah-to situation.  The numbers for these architectures are hardcoded here because we build the product on an old distribution for compatibility reasons, and it's often using outdate kernel include headers (so new `__NR_*` defines aren't there).  Of course, once a syscall number is chosen, its number is carved in stone and will never change, so it's fine to hardcode it.

I don't understand (2). Are you suggesting adding a dependency on systemd headers?

The system call numbers are defined in either system-specific headers (e.g. on ARM arches), or in a DSL that's read during build to generate header files (e.g. [on x32, x86 and x86_64](https://elixir.bootlin.com/linux/v5.3.2/source/arch/x86/entry/syscalls/syscall_64.tbl)); this second approach seems to be gaining traction.  In other words, it's not easily greppable in the Linux source tree unless you know what you're looking for.

Reply:
Question 2 should've been a question actually, my bad. I was wondering if there was an official file where `PAL_COPY_FILE_RANGE_SYSCALL` was defined so you didn't have to redefine it here. If it's not possible, or you don't think it's convenient, feel free to ignore it.

Reply:
I like suggestion (1) above.  Can we switch to that?  I realize it's a minor difference, but I think the flow of saying "if we can get the value from the system, do so, otherwise use these hardcoded values" is better than "use these hardcoded values and if we're on some platform where we don't know the hardcoded value then try to get it from the system".